### DISCUSSION TITLE
Right way to retrieve GraphQLServletContext using graphql-java 17.0+

### DISCUSSION THREAD

**Discussion by pixel13** [graphql-java-kickstart/graphql-spring-boot#733] [Q&A]
Which is the right way to retrieve the `GraphQLServletContext` inside a resolver, now that the `getContext()` method in `DataFetchingEnvironment` has been deprecated (https://github.com/graphql-java/graphql-java/pull/2368)?

**Comment by abhi170392** [comment]
@oliemansm / @vojtapol Can you provide help?

**Comment by abhi170392** [comment]
@pixel13 Did you found a way to deal this? If yes can you provide details?

**Reply by pixel13** [reply]
No, I'm currently using the deprecated method `getContext()`, I didn't find any other way to do it.

**Comment by sapinheiro** [comment]
Also having this problem

**Reply by sapinheiro** [reply]
@oliemansm @vojtapol

**Comment by oliemansm** [comment]
The ground work on servlet has been done. Expect to be able to release a new version soon.

**Reply by sapinheiro** [reply]
Does that mean we should just be using `getContext` until the new release? Or is there a workaround (more specific question: https://github.com/graphql-java-kickstart/graphql-java-tools/discussions/627)

**Reply by oliemansm** [reply]
Yes
